### PR TITLE
feat(adapters): upgrade LangChain packages to 1.x

### LINKED ISSUE #1470: TypeScript: update LangChain packages to the latest version
The TypeScript package depends on outdated LangChain versions (`@langchain/core` ^0.2/^0.3, `@langchain/community` ^0.2/^0.3 in `typescript/package.json`), while LangChain JS 1.x is out and Python already uses langchain 1.x. Upgrade the `@langchain/*` packages to the latest versions, resolve the duplicate version ranges, and adapt the LangChain adapter (`typescript/src/adapters/langchain`) to any breaking API changes.

### PR DESCRIPTION
**Pull request by J-Srinivasalu**
- Upgrade @langchain/core from ^0.2.27 to ^1.1.49
- Upgrade @langchain/community from ^0.2.28 to ^1.1.29
- Upgrade @langchain/langgraph from ^0.4.9 to ^1.4.2
- Upgrade langchain from ^0.3.36 to ^1.4.5
- Add comprehensive message conversion utilities in utils.ts
- Refactor chat.ts to use new conversion functions
- Add extensive test coverage for message conversions
- Align with Python implementation using langchain 1.x

Fixes #1470

### Which issue(s) does this pull-request address?

Closes: #1470

### Description

This PR upgrades the TypeScript LangChain packages from versions 0.2/0.3 to 1.x to align with the latest LangChain JS release and match the Python implementation which already uses langchain 1.x.

**Key Changes:**
- Upgraded all `@langchain/*` packages to their latest 1.x versions
- Created comprehensive message conversion utilities (`utils.ts`) for bidirectional conversion between BeeAI and LangChain message formats
- Refactored `chat.ts` to use the new conversion functions, removing inline conversion logic
- Added 11 unit tests covering all message conversion scenarios including edge cases

**Testing:**
- All unit tests pass (11/11 in utils.test.ts)
- Integration tested with Ollama + llama3.2:1b
- TypeScript compilation successful
- No breaking changes to existing BeeAI code

#### General
- [x] I have read the appropriate contributor guide: [TypeScript](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
- [x] I have signed off on my commit: [TypeScript instructions](https://github.com/i-am-bee/beeai-framework/blob/main/typescript/CONTRIBUTING.md#developer-certificate-of-origin-dco)
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Appropriate label(s) added to PR: `TypeScript` for TypeScript changes

#### Code quality checks
- [x] Code quality checks pass: `mise check` (`mise fix` to auto-fix)

#### Testing
- [x] Unit tests pass: `mise test:unit`
- [ ] E2E tests pass: `mise test:e2e` (not applicable for this change)
- [x] Tests are included (for bug fixes or new features)

#### Documentation
- [ ] Documentation is updated (no documentation changes needed)
- [ ] Embedme embeds code examples in docs (not applicable)

### CODE REVIEW

File: typescript/src/adapters/langchain/backend/utils.ts [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+  let content: string;
+  if (typeof message.content === "string" && message.content) {
+    content = message.content;
+  } else if (message.content) {
+    content = JSON.stringify(message.content);
+  } else if ("artifact" in message && message.artifact) {
+    content = JSON.stringify(message.artifact);
+  } else {
+    content = "";
+  }
```
[the remark this answers is not in the document]
Comment (J-Srinivasalu) [reply typescript/src/adapters/langchain/backend/utils.ts:284]: Fixed in commit d625d2fa. Changed from truthiness checks to explicit null/undefined checks to preserve valid falsy values like 0, false, and empty strings. Tool results now correctly handle all valid outputs.

File: typescript/src/adapters/langchain/backend/utils.ts [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+  if (isString(input)) {
+    const parsed = parseBrokenJson(input, { pair: ["{", "}"] });
+    if (isPlainObject(parsed)) {
+      return parsed;
+    }
+  }
+
+  return {};
+}
```
[the remark this answers is not in the document]
Comment (J-Srinivasalu) [reply typescript/src/adapters/langchain/backend/utils.ts:110]: Fixed in commit d625d2fa. Wrapped parseBrokenJson in try-catch block with warning log. Malformed JSON now fails gracefully and returns empty object instead of crashing the conversion process.

File: typescript/src/adapters/langchain/backend/utils.ts [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    if (msg instanceof SystemMessage) {
+      output.push(new LCSystemMessage({ content, id: msg.id }));
+      continue;
+    }
+
+    output.push(new LCUserMessage({ content, id: msg.id }));
+  }
+
+  return output;
+}
```
[the remark this answers is not in the document]
Comment (J-Srinivasalu) [reply typescript/src/adapters/langchain/backend/utils.ts:354]: Fixed in commit d625d2fa. Moved toLCContent() calls inline for each message type. This eliminates redundant computation for ToolMessages and avoids unnecessary warning logs.

File: typescript/src/adapters/langchain/backend/utils.ts [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+      return new SystemMessage(toBeeAITextParts(msg.content), msg.response_metadata, msg.id);
+    }
+
+    if (msg instanceof LCToolMessage) {
+      return new ToolMessage(toBeeAIToolResult(msg), msg.response_metadata, msg.id);
+    }
+
+    throw new ValueError(`Unsupported message type: ${msg.constructor.name}`);
+  });
+}
```
[the remark this answers is not in the document]
Comment (J-Srinivasalu) [reply typescript/src/adapters/langchain/backend/utils.ts:389]: Fixed in commit d625d2fa. Updated to use .type property (LangChain 1.x standard) instead of deprecated _getType(). This ensures readable error messages in both development and minified production builds.
